application/src/main/java/org/thingsboard/server/actors/ActorSystemContext.java

```diff
@@ -639,6 +639,18 @@ public void init() {
     @Getter
     private long cfCalculationResultTimeout;
 
+    @Value("${mqtt.client.retransmission.max_attempts:3}")
+    @Getter
+    private int mqttClientRetransmissionMaxAttempts;
```

Comment:
Move this 3 parameters to a separate @Component and use them as one object here

Reply:
I agree that it would be better to use a single autowired object here, but I suggest keeping it as is because:

- The `@Component` approach was already used in the PE for configuring the MQTT integration client.  
- It makes sense to reuse this component in two places - MQTT integration and the MQTT rule node - to avoid having two identical classes in two different places
- I did not find a suitable module in which to place this component so that it’s a `@Component` available to both integration and rule engine (except `common/queue`, but placing MQTT client settings in a queue-dedicated module doesn’t feel appropriate).

So we either duplicate essentially the same class twice, or place the component into a strange module, or leave it as is.

Reply:
Sometimes It is better to duplicate then to leave as is. CE code quality don't need to suffer because we have identical lines in PE. Please change the code as i have requested or it will not get merged